```
Error running mint automation
+ mkdir -p 9858-ed079a8/gopath/src/github.com/minio
+ git -C 9858-ed079a8/gopath/src/github.com/minio clone --quiet https://github.com/minio/minio.git
+ git -C 9858-ed079a8/gopath/src/github.com/minio/minio remote add minio https://github.com/minio/minio.git
+ git -C 9858-ed079a8/gopath/src/github.com/minio/minio fetch --quiet minio pull/9858/head:pr9858
+ git -C 9858-ed079a8/gopath/src/github.com/minio/minio checkout --quiet pr9858
+ GO111MODULE=on
+ GOPATH=/home/mint-auto/mint-auto/9858-ed079a8/gopath
+ make -C 9858-ed079a8/gopath/src/github.com/minio/minio --quiet
go: downloading gopkg.in/jcmturner/gokrb5.v7 v7.3.0
go: downloading google.golang.org/api v0.5.0
go: downloading github.com/minio/minio-go/v6 v6.0.58-0.20200612001654-a57fec8037ec
go: downloading github.com/Azure/azure-storage-blob-go v0.8.0
go: downloading github.com/minio/highwayhash v1.0.0
go: downloading github.com/fatih/color v1.7.0
go: downloading github.com/colinmarc/hdfs/v2 v2.1.1
go: downloading github.com/miekg/dns v1.1.8
go: extracting github.com/fatih/color v1.7.0
go: downloading github.com/go-sql-driver/mysql v1.5.0
go: extracting github.com/minio/highwayhash v1.0.0
go: extracting github.com/Azure/azure-storage-blob-go v0.8.0
go: extracting github.com/miekg/dns v1.1.8
go: downloading golang.org/x/sys v0.0.0-20200610111108-226ff32320da
go: downloading github.com/gorilla/mux v1.7.0
go: extracting github.com/go-sql-driver/mysql v1.5.0
go: downloading github.com/cespare/xxhash v1.1.0
go: extracting github.com/minio/minio-go/v6 v6.0.58-0.20200612001654-a57fec8037ec
go: extracting github.com/colinmarc/hdfs/v2 v2.1.1
go: downloading cloud.google.com/go v0.39.0
go: downloading github.com/klauspost/reedsolomon v1.9.9
go: downloading github.com/dgrijalva/jwt-go v3.2.0+incompatible
go: extracting github.com/cespare/xxhash v1.1.0
go: downloading github.com/cespare/xxhash/v2 v2.1.1
go: extracting github.com/gorilla/mux v1.7.0
go: downloading github.com/minio/simdjson-go v0.1.5-0.20200303142138-b17fe061ea37
go: extracting golang.org/x/sys v0.0.0-20200610111108-226ff32320da
go: extracting github.com/cespare/xxhash/v2 v2.1.1
go: downloading github.com/klauspost/compress v1.10.3
go: extracting github.com/dgrijalva/jwt-go v3.2.0+incompatible
go: downloading github.com/coreos/etcd v3.3.12+incompatible
go: extracting github.com/klauspost/reedsolomon v1.9.9
go: downloading github.com/lib/pq v1.1.1
go: downloading github.com/json-iterator/go v1.1.10
go: extracting github.com/minio/simdjson-go v0.1.5-0.20200303142138-b17fe061ea37
go: extracting github.com/json-iterator/go v1.1.10
go: extracting github.com/lib/pq v1.1.1
go: downloading github.com/dchest/siphash v1.2.1
go: downloading github.com/klauspost/readahead v1.3.1
go: extracting github.com/klauspost/readahead v1.3.1
go: downloading golang.org/x/net v0.0.0-20200602114024-627f9648deb9
go: extracting github.com/dchest/siphash v1.2.1
go: downloading github.com/Shopify/sarama v1.24.1
go: extracting github.com/coreos/etcd v3.3.12+incompatible
go: extracting cloud.google.com/go v0.39.0
go: extracting github.com/Shopify/sarama v1.24.1
go: downloading github.com/golang/protobuf v1.3.2
go: downloading github.com/shirou/gopsutil v2.20.3-0.20200314133625-53cec6b37e6a+incompatible
go: extracting golang.org/x/net v0.0.0-20200602114024-627f9648deb9
go: extracting github.com/golang/protobuf v1.3.2
go: downloading github.com/tinylib/msgp v1.1.1
go: downloading google.golang.org/grpc v1.22.0
go: downloading github.com/gorilla/rpc v1.2.0
go: extracting github.com/tinylib/msgp v1.1.1
go: downloading github.com/minio/cli v1.22.0
go: extracting github.com/shirou/gopsutil v2.20.3-0.20200314133625-53cec6b37e6a+incompatible
go: downloading github.com/klauspost/pgzip v1.2.1
go: extracting github.com/gorilla/rpc v1.2.0
go: downloading github.com/fatih/structs v1.1.0
go: extracting github.com/minio/cli v1.22.0
go: downloading github.com/rs/cors v1.7.0
go: extracting github.com/fatih/structs v1.1.0
go: extracting github.com/klauspost/pgzip v1.2.1
go: downloading github.com/nats-io/stan.go v0.4.5
go: downloading gopkg.in/ldap.v3 v3.0.3
go: extracting google.golang.org/grpc v1.22.0
go: downloading github.com/eclipse/paho.mqtt.golang v1.2.0
go: extracting github.com/nats-io/stan.go v0.4.5
go: downloading gopkg.in/yaml.v2 v2.2.8
go: extracting github.com/eclipse/paho.mqtt.golang v1.2.0
go: extracting github.com/rs/cors v1.7.0
go: extracting gopkg.in/ldap.v3 v3.0.3
go: downloading github.com/willf/bloom v2.0.3+incompatible
go: downloading google.golang.org/genproto v0.0.0-20190513181449-d00d292a067c
go: downloading github.com/googleapis/gax-go/v2 v2.0.4
go: extracting github.com/willf/bloom v2.0.3+incompatible
go: downloading github.com/prometheus/client_golang v0.9.3
go: extracting gopkg.in/yaml.v2 v2.2.8
go: downloading github.com/ncw/directio v1.0.5
go: extracting github.com/googleapis/gax-go/v2 v2.0.4
go: downloading golang.org/x/crypto v0.0.0-20200604202706-70a84ac30bf9
go: extracting github.com/klauspost/compress v1.10.3
go: extracting github.com/ncw/directio v1.0.5
go: downloading github.com/valyala/tcplisten v0.0.0-20161114210144-ceec8f93295a
go: extracting golang.org/x/crypto v0.0.0-20200604202706-70a84ac30bf9
go: downloading github.com/mitchellh/go-homedir v1.1.0
go: downloading github.com/xdg/scram v0.0.0-20180814205039-7eeb5667e42c
go: extracting github.com/prometheus/client_golang v0.9.3
go: downloading github.com/google/uuid v1.1.1
go: downloading gopkg.in/asn1-ber.v1 v1.0.0-20181015200546-f715ec2f112d
go: extracting github.com/mitchellh/go-homedir v1.1.0
go: downloading github.com/cheggaaa/pb v1.0.28
go: extracting github.com/valyala/tcplisten v0.0.0-20161114210144-ceec8f93295a
go: extracting github.com/xdg/scram v0.0.0-20180814205039-7eeb5667e42c
go: downloading github.com/beorn7/perks v1.0.0
go: downloading github.com/streadway/amqp v0.0.0-20190404075320-75d898a42a94
go: extracting github.com/beorn7/perks v1.0.0
go: extracting github.com/cheggaaa/pb v1.0.28
go: extracting github.com/google/uuid v1.1.1
go: extracting gopkg.in/asn1-ber.v1 v1.0.0-20181015200546-f715ec2f112d
go: downloading github.com/elazarl/go-bindata-assetfs v1.0.0
go: downloading github.com/mattn/go-runewidth v0.0.4
go: downloading gopkg.in/olivere/elastic.v5 v5.0.80
go: extracting github.com/streadway/amqp v0.0.0-20190404075320-75d898a42a94
go: downloading github.com/gogo/protobuf v1.2.1
go: extracting github.com/elazarl/go-bindata-assetfs v1.0.0
go: downloading github.com/prometheus/client_model v0.0.0-20190129233127-fd36f4220a90
go: extracting github.com/mattn/go-runewidth v0.0.4
go: downloading github.com/minio/parquet-go v0.0.0-20200414234858-838cfa8aae61
go: extracting github.com/prometheus/client_model v0.0.0-20190129233127-fd36f4220a90
go: extracting github.com/minio/parquet-go v0.0.0-20200414234858-838cfa8aae61
go: extracting gopkg.in/olivere/elastic.v5 v5.0.80
go: downloading github.com/spaolacci/murmur3 v1.1.0
go: extracting github.com/spaolacci/murmur3 v1.1.0
go: downloading git.apache.org/thrift.git v0.13.0
go: downloading github.com/pierrec/lz4 v2.4.0+incompatible
go: extracting google.golang.org/genproto v0.0.0-20190513181449-d00d292a067c
go: extracting google.golang.org/api v0.5.0
go: downloading github.com/minio/md5-simd v1.1.0
go: downloading github.com/prometheus/common v0.4.0
go: downloading golang.org/x/oauth2 v0.0.0-20190402181905-9f3314589c9a
go: downloading github.com/mattn/go-isatty v0.0.8
go: extracting github.com/gogo/protobuf v1.2.1
go: downloading github.com/philhofer/fwd v1.0.0
go: extracting github.com/prometheus/common v0.4.0
go: extracting github.com/minio/md5-simd v1.1.0
go: downloading github.com/Azure/azure-pipeline-go v0.2.1
go: extracting github.com/mattn/go-isatty v0.0.8
go: downloading github.com/willf/bitset v1.1.10
go: extracting golang.org/x/oauth2 v0.0.0-20190402181905-9f3314589c9a
go: downloading github.com/minio/sha256-simd v0.1.1
go: downloading github.com/rjeczalik/notify v0.9.2
go: extracting github.com/philhofer/fwd v1.0.0
go: downloading github.com/mailru/easyjson v0.0.0-20180730094502-03f2033d19d5
go: extracting github.com/Azure/azure-pipeline-go v0.2.1
go: extracting github.com/willf/bitset v1.1.10
go: downloading github.com/bcicen/jstream v0.0.0-20190220045926-16c1f8af81c2
go: downloading github.com/golang/snappy v0.0.1
go: extracting github.com/minio/sha256-simd v0.1.1
go: downloading github.com/xdg/stringprep v1.0.0
go: extracting github.com/rjeczalik/notify v0.9.2
go: downloading github.com/hashicorp/vault/api v1.0.4
go: extracting github.com/golang/snappy v0.0.1
go: extracting github.com/xdg/stringprep v1.0.0
go: downloading golang.org/x/text v0.3.2
go: extracting github.com/mailru/easyjson v0.0.0-20180730094502-03f2033d19d5
go: downloading github.com/nats-io/nats.go v1.9.1
go: extracting github.com/hashicorp/vault/api v1.0.4
go: extracting git.apache.org/thrift.git v0.13.0
go: downloading github.com/hashicorp/go-multierror v1.0.0
go: extracting github.com/nats-io/nats.go v1.9.1
go: downloading github.com/dustin/go-humanize v1.0.0
go: extracting github.com/hashicorp/go-multierror v1.0.0
go: downloading github.com/nats-io/nuid v1.0.1
go: extracting github.com/pierrec/lz4 v2.4.0+incompatible
go: extracting github.com/nats-io/nuid v1.0.1
go: extracting github.com/bcicen/jstream v0.0.0-20190220045926-16c1f8af81c2
go: downloading github.com/gomodule/redigo v2.0.0+incompatible
go: extracting github.com/dustin/go-humanize v1.0.0
go: downloading github.com/mattn/go-colorable v0.1.4
go: downloading github.com/nats-io/jwt v0.3.2
go: extracting github.com/gomodule/redigo v2.0.0+incompatible
go: downloading github.com/eapache/go-xerial-snappy v0.0.0-20180814174437-776d5712da21
go: extracting github.com/mattn/go-colorable v0.1.4
go: downloading github.com/beevik/ntp v0.2.0
go: extracting github.com/nats-io/jwt v0.3.2
go: downloading github.com/rcrowley/go-metrics v0.0.0-20190704165056-9c2d0518ed81
go: extracting github.com/eapache/go-xerial-snappy v0.0.0-20180814174437-776d5712da21
go: downloading github.com/davecgh/go-spew v1.1.1
go: downloading github.com/eapache/go-resiliency v1.2.0
go: extracting github.com/beevik/ntp v0.2.0
go: downloading github.com/hashicorp/hcl v1.0.0
go: extracting github.com/rcrowley/go-metrics v0.0.0-20190704165056-9c2d0518ed81
go: downloading github.com/hashicorp/vault/sdk v0.1.13
go: extracting github.com/eapache/go-resiliency v1.2.0
go: extracting github.com/davecgh/go-spew v1.1.1
go: extracting github.com/hashicorp/hcl v1.0.0
go: downloading github.com/modern-go/concurrent v0.0.0-20180306012644-bacd9c7ef1dd
go: downloading github.com/matttproud/golang_protobuf_extensions v1.0.1
go: downloading go.opencensus.io v0.21.0
go: extracting github.com/modern-go/concurrent v0.0.0-20180306012644-bacd9c7ef1dd
go: extracting github.com/matttproud/golang_protobuf_extensions v1.0.1
go: extracting github.com/hashicorp/vault/sdk v0.1.13
go: downloading github.com/djherbis/atime v1.0.0
go: extracting go.opencensus.io v0.21.0
go: extracting github.com/djherbis/atime v1.0.0
go: downloading github.com/inconshreveable/go-update v0.0.0-20160112193335-8152e7eb6ccf
go: downloading github.com/prometheus/procfs v0.0.0-20190507164030-5867b95ac084
go: extracting golang.org/x/text v0.3.2
go: extracting github.com/inconshreveable/go-update v0.0.0-20160112193335-8152e7eb6ccf
go: downloading github.com/hashicorp/golang-lru v0.5.1
go: extracting github.com/hashicorp/golang-lru v0.5.1
go: downloading github.com/coredns/coredns v1.4.0
go: extracting github.com/prometheus/procfs v0.0.0-20190507164030-5867b95ac084
go: downloading github.com/hashicorp/go-retryablehttp v0.5.4
go: downloading github.com/mitchellh/mapstructure v1.1.2
go: extracting github.com/hashicorp/go-retryablehttp v0.5.4
go: downloading github.com/tidwall/sjson v1.0.4
go: extracting github.com/mitchellh/mapstructure v1.1.2
go: downloading github.com/modern-go/reflect2 v1.0.1
go: extracting github.com/coredns/coredns v1.4.0
go: extracting github.com/tidwall/sjson v1.0.4
go: downloading github.com/hashicorp/go-cleanhttp v0.5.1
go: extracting github.com/modern-go/reflect2 v1.0.1
go: downloading github.com/tidwall/gjson v1.3.5
go: downloading gopkg.in/square/go-jose.v2 v2.3.1
go: extracting github.com/hashicorp/go-cleanhttp v0.5.1
go: downloading github.com/nats-io/nkeys v0.1.3
go: extracting github.com/tidwall/gjson v1.3.5
go: downloading github.com/hashicorp/go-sockaddr v1.0.2
go: extracting github.com/nats-io/nkeys v0.1.3
go: downloading github.com/tidwall/match v1.0.1
go: extracting github.com/hashicorp/go-sockaddr v1.0.2
go: downloading github.com/alecthomas/participle v0.2.1
go: extracting github.com/tidwall/match v1.0.1
go: extracting gopkg.in/square/go-jose.v2 v2.3.1
go: extracting github.com/alecthomas/participle v0.2.1
go: downloading github.com/minio/sio v0.2.0
go: extracting github.com/minio/sio v0.2.0
go: downloading golang.org/x/time v0.0.0-20190308202827-9d24e82272b4
go: extracting golang.org/x/time v0.0.0-20190308202827-9d24e82272b4
go: downloading github.com/pkg/errors v0.8.1
go: extracting github.com/pkg/errors v0.8.1
go: downloading github.com/nsqio/go-nsq v1.0.7
go: extracting github.com/nsqio/go-nsq v1.0.7
go: downloading github.com/hashicorp/go-rootcerts v1.0.1
go: extracting github.com/hashicorp/go-rootcerts v1.0.1
go: downloading github.com/montanaflynn/stats v0.5.0
go: extracting github.com/montanaflynn/stats v0.5.0
go: extracting gopkg.in/jcmturner/gokrb5.v7 v7.3.0
go: downloading github.com/tidwall/pretty v1.0.0
go: downloading github.com/gorilla/handlers v1.4.0
go: downloading github.com/eapache/queue v1.1.0
go: downloading github.com/mattn/go-ieproxy v0.0.0-20190805055040-f9202b1cfdeb
go: downloading github.com/hashicorp/go-uuid v1.0.1
go: downloading gopkg.in/jcmturner/dnsutils.v1 v1.0.1
go: downloading github.com/hashicorp/errwrap v1.0.0
go: downloading github.com/ryanuber/go-glob v1.0.0
go: downloading github.com/klauspost/cpuid v1.3.0
go: downloading github.com/secure-io/sio-go v0.3.0
go: extracting github.com/tidwall/pretty v1.0.0
go: downloading gopkg.in/jcmturner/rpc.v1 v1.1.0
go: extracting github.com/eapache/queue v1.1.0
go: extracting github.com/gorilla/handlers v1.4.0
go: extracting github.com/hashicorp/errwrap v1.0.0
go: extracting github.com/hashicorp/go-uuid v1.0.1
go: downloading gopkg.in/jcmturner/goidentity.v3 v3.0.0
go: extracting gopkg.in/jcmturner/dnsutils.v1 v1.0.1
go: downloading github.com/jcmturner/gofork v0.0.0-20190328161633-dc7c13fece03
go: downloading gopkg.in/jcmturner/aescts.v1 v1.0.1
go: extracting github.com/ryanuber/go-glob v1.0.0
go: extracting github.com/secure-io/sio-go v0.3.0
go: downloading gopkg.in/ini.v1 v1.57.0
go: extracting github.com/mattn/go-ieproxy v0.0.0-20190805055040-f9202b1cfdeb
go: extracting gopkg.in/jcmturner/rpc.v1 v1.1.0
go: extracting gopkg.in/jcmturner/goidentity.v3 v3.0.0
go: extracting gopkg.in/jcmturner/aescts.v1 v1.0.1
go: extracting github.com/jcmturner/gofork v0.0.0-20190328161633-dc7c13fece03
go: extracting gopkg.in/ini.v1 v1.57.0
go: extracting github.com/klauspost/cpuid v1.3.0
go: finding gopkg.in/jcmturner/gokrb5.v7 v7.3.0
go: finding github.com/dustin/go-humanize v1.0.0
go: finding github.com/minio/minio-go/v6 v6.0.58-0.20200612001654-a57fec8037ec
go: finding github.com/colinmarc/hdfs/v2 v2.1.1
go: finding google.golang.org/api v0.5.0
go: finding github.com/minio/sha256-simd v0.1.1
go: finding github.com/minio/cli v1.22.0
go: finding github.com/Azure/azure-pipeline-go v0.2.1
go: finding cloud.google.com/go v0.39.0
go: finding github.com/minio/highwayhash v1.0.0
go: finding github.com/dgrijalva/jwt-go v3.2.0+incompatible
go: finding github.com/gorilla/mux v1.7.0
go: finding github.com/Azure/azure-storage-blob-go v0.8.0
go: finding github.com/mattn/go-isatty v0.0.8
go: finding github.com/montanaflynn/stats v0.5.0
go: finding github.com/mattn/go-colorable v0.1.4
go: finding github.com/json-iterator/go v1.1.10
go: finding github.com/fatih/color v1.7.0
go: finding github.com/valyala/tcplisten v0.0.0-20161114210144-ceec8f93295a
go: finding github.com/rjeczalik/notify v0.9.2
go: finding github.com/secure-io/sio-go v0.3.0
go: finding github.com/shirou/gopsutil v2.20.3-0.20200314133625-53cec6b37e6a+incompatible
go: finding github.com/ncw/directio v1.0.5
go: finding golang.org/x/crypto v0.0.0-20200604202706-70a84ac30bf9
go: finding golang.org/x/net v0.0.0-20200602114024-627f9648deb9
go: finding golang.org/x/sys v0.0.0-20200610111108-226ff32320da
go: finding gopkg.in/jcmturner/dnsutils.v1 v1.0.1
go: finding github.com/hashicorp/go-uuid v1.0.1
go: finding github.com/jcmturner/gofork v0.0.0-20190328161633-dc7c13fece03
go: finding github.com/mitchellh/go-homedir v1.1.0
go: finding golang.org/x/oauth2 v0.0.0-20190402181905-9f3314589c9a
go: finding gopkg.in/jcmturner/goidentity.v3 v3.0.0
go: finding github.com/minio/md5-simd v1.1.0
go: finding google.golang.org/grpc v1.22.0
go: finding gopkg.in/ini.v1 v1.57.0
go: finding gopkg.in/jcmturner/aescts.v1 v1.0.1
go: finding github.com/mattn/go-ieproxy v0.0.0-20190805055040-f9202b1cfdeb
go: finding gopkg.in/jcmturner/rpc.v1 v1.1.0
go: finding github.com/golang/protobuf v1.3.2
go: finding github.com/googleapis/gax-go/v2 v2.0.4
go: finding google.golang.org/genproto v0.0.0-20190513181449-d00d292a067c
go: finding go.opencensus.io v0.21.0
go: finding github.com/modern-go/concurrent v0.0.0-20180306012644-bacd9c7ef1dd
go: finding github.com/modern-go/reflect2 v1.0.1
go: finding golang.org/x/text v0.3.2
go: finding github.com/klauspost/cpuid v1.3.0
go: finding github.com/cespare/xxhash/v2 v2.1.1
go: finding github.com/hashicorp/golang-lru v0.5.1
go: finding github.com/cheggaaa/pb v1.0.28
go: finding github.com/mattn/go-runewidth v0.0.4
go: finding github.com/coreos/etcd v3.3.12+incompatible
go: finding github.com/gogo/protobuf v1.2.1
go: finding github.com/dchest/siphash v1.2.1
go: finding github.com/djherbis/atime v1.0.0
go: finding github.com/elazarl/go-bindata-assetfs v1.0.0
go: finding github.com/google/uuid v1.1.1
go: finding github.com/gorilla/handlers v1.4.0
go: finding github.com/gorilla/rpc v1.2.0
go: finding github.com/inconshreveable/go-update v0.0.0-20160112193335-8152e7eb6ccf
go: finding github.com/klauspost/compress v1.10.3
go: finding github.com/klauspost/readahead v1.3.1
go: finding github.com/klauspost/reedsolomon v1.9.9
go: finding github.com/miekg/dns v1.1.8
go: finding github.com/coredns/coredns v1.4.0
go: finding gopkg.in/ldap.v3 v3.0.3
go: finding gopkg.in/asn1-ber.v1 v1.0.0-20181015200546-f715ec2f112d
go: finding github.com/Shopify/sarama v1.24.1
go: finding github.com/davecgh/go-spew v1.1.1
go: finding github.com/eapache/go-resiliency v1.2.0
go: finding github.com/eapache/go-xerial-snappy v0.0.0-20180814174437-776d5712da21
go: finding github.com/golang/snappy v0.0.1
go: finding github.com/eapache/queue v1.1.0
go: finding github.com/pierrec/lz4 v2.4.0+incompatible
go: finding github.com/rcrowley/go-metrics v0.0.0-20190704165056-9c2d0518ed81
go: finding github.com/eclipse/paho.mqtt.golang v1.2.0
go: finding github.com/go-sql-driver/mysql v1.5.0
go: finding github.com/gomodule/redigo v2.0.0+incompatible
go: finding github.com/lib/pq v1.1.1
go: finding github.com/nats-io/nats.go v1.9.1
go: finding github.com/nats-io/jwt v0.3.2
go: finding github.com/nats-io/nkeys v0.1.3
go: finding github.com/nats-io/nuid v1.0.1
go: finding github.com/nats-io/stan.go v0.4.5
go: finding github.com/nsqio/go-nsq v1.0.7
go: finding github.com/pkg/errors v0.8.1
go: finding github.com/streadway/amqp v0.0.0-20190404075320-75d898a42a94
go: finding github.com/xdg/scram v0.0.0-20180814205039-7eeb5667e42c
go: finding github.com/xdg/stringprep v1.0.0
go: finding gopkg.in/olivere/elastic.v5 v5.0.80
go: finding github.com/mailru/easyjson v0.0.0-20180730094502-03f2033d19d5
go: finding github.com/hashicorp/vault/api v1.0.4
go: finding github.com/hashicorp/errwrap v1.0.0
go: finding github.com/hashicorp/go-cleanhttp v0.5.1
go: finding github.com/hashicorp/go-multierror v1.0.0
go: finding github.com/hashicorp/go-retryablehttp v0.5.4
go: finding github.com/hashicorp/go-rootcerts v1.0.1
go: finding github.com/hashicorp/hcl v1.0.0
go: finding github.com/hashicorp/vault/sdk v0.1.13
go: finding github.com/hashicorp/go-sockaddr v1.0.2
go: finding github.com/ryanuber/go-glob v1.0.0
go: finding github.com/mitchellh/mapstructure v1.1.2
go: finding golang.org/x/time v0.0.0-20190308202827-9d24e82272b4
go: finding gopkg.in/square/go-jose.v2 v2.3.1
go: finding github.com/minio/sio v0.2.0
go: finding github.com/beevik/ntp v0.2.0
go: finding github.com/fatih/structs v1.1.0
go: finding gopkg.in/yaml.v2 v2.2.8
go: finding github.com/klauspost/pgzip v1.2.1
go: finding github.com/bcicen/jstream v0.0.0-20190220045926-16c1f8af81c2
go: finding github.com/alecthomas/participle v0.2.1
go: finding github.com/minio/simdjson-go v0.1.5-0.20200303142138-b17fe061ea37
go: finding github.com/minio/parquet-go v0.0.0-20200414234858-838cfa8aae61
go: finding git.apache.org/thrift.git v0.13.0
go: finding github.com/tidwall/gjson v1.3.5
go: finding github.com/tidwall/match v1.0.1
go: finding github.com/tidwall/pretty v1.0.0
go: finding github.com/tidwall/sjson v1.0.4
go: finding github.com/prometheus/client_golang v0.9.3
go: finding github.com/beorn7/perks v1.0.0
go: finding github.com/prometheus/client_model v0.0.0-20190129233127-fd36f4220a90
go: finding github.com/prometheus/common v0.4.0
go: finding github.com/matttproud/golang_protobuf_extensions v1.0.1
go: finding github.com/prometheus/procfs v0.0.0-20190507164030-5867b95ac084
go: finding github.com/rs/cors v1.7.0
go: finding github.com/tinylib/msgp v1.1.1
go: finding github.com/philhofer/fwd v1.0.0
go: finding github.com/willf/bloom v2.0.3+incompatible
go: finding github.com/spaolacci/murmur3 v1.1.0
go: finding github.com/willf/bitset v1.1.10
# github.com/minio/minio/cmd/rest
cmd/rest/client.go:176:10: c.HealthCheckPath undefined (type *Client has no field or method HealthCheckPath)
make: *** [build] Error 2

```

--- a/cmd/handler-utils.go
+++ b/cmd/handler-utils.go
@@ -401,7 +401,7 @@ func errorResponseHandler(w http.ResponseWriter, r *http.Request) {
 		writeErrorResponseString(r.Context(), w, APIError{
 			Code:           "XMinioPeerVersionMismatch",
 			Description:    desc,
-			HTTPStatusCode: http.StatusBadRequest,
+			HTTPStatusCode: http.StatusUpgradeRequired,
 		}, r.URL)
 	case strings.HasPrefix(r.URL.Path, storageRESTPrefix):
 		desc := fmt.Sprintf("Expected 'storage' API version '%s', instead found '%s', please upgrade the servers",
@@ -409,7 +409,7 @@ func errorResponseHandler(w http.ResponseWriter, r *http.Request) {
 		writeErrorResponseString(r.Context(), w, APIError{
 			Code:           "XMinioStorageVersionMismatch",
 			Description:    desc,
-			HTTPStatusCode: http.StatusBadRequest,
+			HTTPStatusCode: http.StatusUpgradeRequired,
 		}, r.URL)
 	case strings.HasPrefix(r.URL.Path, lockRESTPrefix):
 		desc := fmt.Sprintf("Expected 'lock' API version '%s', instead found '%s', please upgrade the servers",
@@ -417,7 +417,7 @@ func errorResponseHandler(w http.ResponseWriter, r *http.Request) {
 		writeErrorResponseString(r.Context(), w, APIError{
 			Code:           "XMinioLockVersionMismatch",
 			Description:    desc,
-			HTTPStatusCode: http.StatusBadRequest,
+			HTTPStatusCode: http.StatusUpgradeRequired,
 		}, r.URL)
 	case strings.HasPrefix(r.URL.Path, adminPathPrefix):
 		var desc string
@@ -431,7 +431,7 @@ func errorResponseHandler(w http.ResponseWriter, r *http.Request) {
 		writeErrorResponseJSON(r.Context(), w, APIError{
 			Code:           "XMinioAdminVersionMismatch",
 			Description:    desc,
-			HTTPStatusCode: http.StatusBadRequest,
+			HTTPStatusCode: http.StatusUpgradeRequired,
 		}, r.URL)
 	default:
 		desc := fmt.Sprintf("Unknown API request at %s", r.URL.Path)

--- a/cmd/lock-rest-client.go
+++ b/cmd/lock-rest-client.go
@@ -18,11 +18,14 @@ package cmd
 
 import (
 	"bytes"
+	"context"
 	"crypto/tls"
+	"errors"
 	"io"
 	"net/url"
 
 	"github.com/minio/minio/cmd/http"
+	xhttp "github.com/minio/minio/cmd/http"
 	"github.com/minio/minio/cmd/logger"
 	"github.com/minio/minio/cmd/rest"
 	"github.com/minio/minio/pkg/dsync"
@@ -156,7 +159,14 @@ func newlockRESTClient(endpoint Endpoint) *lockRESTClient {
 	if err != nil {
 		logger.Fatal(err, "Unable to create lock rest client")
 	}
-	restClient.HealthCheckPath = "/"
+	restClient.HealthCheckFn = func() bool {
+		ctx, cancel := context.WithTimeout(GlobalContext, restClient.HealthCheckTimeout)
+		respBody, err := restClient.CallWithContext(ctx, lockRESTMethodHealth, nil, nil, -1)
+		xhttp.DrainBody(respBody)
+		cancel()
+		var ne *rest.NetworkError
+		return !errors.Is(err, context.DeadlineExceeded) && !errors.As(err, &ne)
+	}
 
 	return &lockRESTClient{endpoint: endpoint, restClient: restClient}
 }

--- a/cmd/lock-rest-server-common.go
+++ b/cmd/lock-rest-server-common.go
@@ -27,6 +27,7 @@ const (
 )
 
 const (
+	lockRESTMethodHealth  = "/health"
 	lockRESTMethodLock    = "/lock"
 	lockRESTMethodRLock   = "/rlock"
 	lockRESTMethodUnlock  = "/unlock"

--- a/cmd/lock-rest-server.go
+++ b/cmd/lock-rest-server.go
@@ -78,6 +78,11 @@ func getLockArgs(r *http.Request) (args dsync.LockArgs, err error) {
 	return args, nil
 }
 
+// HealthHandler returns success if request is authenticated.
+func (l *lockRESTServer) HealthHandler(w http.ResponseWriter, r *http.Request) {
+	l.IsValid(w, r)
+}
+
 // LockHandler - Acquires a lock.
 func (l *lockRESTServer) LockHandler(w http.ResponseWriter, r *http.Request) {
 	if !l.IsValid(w, r) {
@@ -345,6 +350,7 @@ func registerLockRESTHandlers(router *mux.Router, endpointZones EndpointZones) {
 			}
 
 			subrouter := router.PathPrefix(path.Join(lockRESTPrefix, endpoint.Path)).Subrouter()
+			subrouter.Methods(http.MethodPost).Path(lockRESTVersionPrefix + lockRESTMethodHealth).HandlerFunc(httpTraceHdrs(lockServer.HealthHandler)).Queries(queries...)
 			subrouter.Methods(http.MethodPost).Path(lockRESTVersionPrefix + lockRESTMethodLock).HandlerFunc(httpTraceHdrs(lockServer.LockHandler)).Queries(queries...)
 			subrouter.Methods(http.MethodPost).Path(lockRESTVersionPrefix + lockRESTMethodRLock).HandlerFunc(httpTraceHdrs(lockServer.RLockHandler)).Queries(queries...)
 			subrouter.Methods(http.MethodPost).Path(lockRESTVersionPrefix + lockRESTMethodUnlock).HandlerFunc(httpTraceHdrs(lockServer.UnlockHandler)).Queries(queries...)

--- a/cmd/peer-rest-client.go
+++ b/cmd/peer-rest-client.go
@@ -21,6 +21,7 @@ import (
 	"context"
 	"crypto/tls"
 	"encoding/gob"
+	"errors"
 	"io"
 	"io/ioutil"
 	"math"
@@ -32,6 +33,7 @@ import (
 
 	"github.com/dustin/go-humanize"
 	"github.com/minio/minio/cmd/http"
+	xhttp "github.com/minio/minio/cmd/http"
 	"github.com/minio/minio/cmd/logger"
 	"github.com/minio/minio/cmd/rest"
 	"github.com/minio/minio/pkg/event"
@@ -881,7 +883,16 @@ func newPeerRESTClient(peer *xnet.Host) (*peerRESTClient, error) {
 	if err != nil {
 		return nil, err
 	}
-	restClient.HealthCheckPath = peerRESTMethodGetLocalDiskIDs
+
+	// Construct a new health function.
+	restClient.HealthCheckFn = func() bool {
+		ctx, cancel := context.WithTimeout(GlobalContext, restClient.HealthCheckTimeout)
+		respBody, err := restClient.CallWithContext(ctx, peerRESTMethodHealth, nil, nil, -1)
+		xhttp.DrainBody(respBody)
+		cancel()
+		var ne *rest.NetworkError
+		return !errors.Is(err, context.DeadlineExceeded) && !errors.As(err, &ne)
+	}
 
 	return &peerRESTClient{host: peer, restClient: restClient}, nil
 }

--- a/cmd/peer-rest-common.go
+++ b/cmd/peer-rest-common.go
@@ -24,6 +24,7 @@ const (
 )
 
 const (
+	peerRESTMethodHealth                = "/health"
 	peerRESTMethodServerInfo            = "/serverinfo"
 	peerRESTMethodDriveOBDInfo          = "/driveobdinfo"
 	peerRESTMethodNetOBDInfo            = "/netobdinfo"

--- a/cmd/peer-rest-server.go
+++ b/cmd/peer-rest-server.go
@@ -729,6 +729,11 @@ func getLocalDiskIDs(z *erasureZones) []string {
 	return ids
 }
 
+// HealthHandler - returns true of health
+func (s *peerRESTServer) HealthHandler(w http.ResponseWriter, r *http.Request) {
+	s.IsValid(w, r)
+}
+
 // GetLocalDiskIDs - Return disk IDs of all the local disks.
 func (s *peerRESTServer) GetLocalDiskIDs(w http.ResponseWriter, r *http.Request) {
 	if !s.IsValid(w, r) {
@@ -1020,6 +1025,7 @@ func (s *peerRESTServer) IsValid(w http.ResponseWriter, r *http.Request) bool {
 func registerPeerRESTHandlers(router *mux.Router) {
 	server := &peerRESTServer{}
 	subrouter := router.PathPrefix(peerRESTPrefix).Subrouter()
+	subrouter.Methods(http.MethodPost).Path(peerRESTVersionPrefix + peerRESTMethodHealth).HandlerFunc(httpTraceHdrs(server.HealthHandler))
 	subrouter.Methods(http.MethodPost).Path(peerRESTVersionPrefix + peerRESTMethodGetLocks).HandlerFunc(httpTraceHdrs(server.GetLocksHandler))
 	subrouter.Methods(http.MethodPost).Path(peerRESTVersionPrefix + peerRESTMethodServerInfo).HandlerFunc(httpTraceHdrs(server.ServerInfoHandler))
 	subrouter.Methods(http.MethodPost).Path(peerRESTVersionPrefix + peerRESTMethodProcOBDInfo).HandlerFunc(httpTraceHdrs(server.ProcOBDInfoHandler))

--- a/cmd/rest/client.go
+++ b/cmd/rest/client.go
@@ -1,5 +1,5 @@
 /*
- * MinIO Cloud Storage, (C) 2018 MinIO, Inc.
+ * MinIO Cloud Storage, (C) 2018-2020 MinIO, Inc.
  *
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.
@@ -56,10 +56,11 @@ func (n *NetworkError) Unwrap() error {
 
 // Client - http based RPC client.
 type Client struct {
-	// HealthCheckPath is the path to test for health.
-	// If left empty the client will not keep track of health.
-	// Calling this can return any http status code/contents.
-	HealthCheckPath string
+	// HealthCheckFn is the function set to test for health.
+	// If not set the client will not keep track of health.
+	// Calling this returns true or false if the target
+	// is online or offline.
+	HealthCheckFn func() bool
 
 	// HealthCheckInterval will be the duration between re-connection attempts
 	// when a call has failed with a network error.
@@ -116,6 +117,18 @@ func (c *Client) CallWithContext(ctx context.Context, method string, values url.
 	}
 
 	if resp.StatusCode != http.StatusOK {
+		// If server returns 412 pre-condition failed, it would
+		// mean that authentication succeeded, but another
+		// side-channel check has failed, we shall take
+		// the client offline in such situations.
+		// generally all implementations should simply return
+		// 403, but in situations where there is a dependency
+		// with the caller to take the client offline purpose
+		// fully it should make sure to respond with '412'
+		// instead, see cmd/storage-rest-server.go for ideas.
+		if resp.StatusCode == http.StatusPreconditionFailed {
+			c.MarkOffline()
+		}
 		defer xhttp.DrainBody(resp.Body)
 		// Limit the ReadAll(), just in case, because of a bug, the server responds with large data.
 		b, err := ioutil.ReadAll(io.LimitReader(resp.Body, c.MaxErrResponseSize))
@@ -157,7 +170,6 @@ func NewClient(url *url.URL, newCustomTransport func() *http.Transport, newAuthT
 		connected:           online,
 
 		MaxErrResponseSize:  4096,
-		HealthCheckPath:     "",
 		HealthCheckInterval: 200 * time.Millisecond,
 		HealthCheckTimeout:  time.Second,
 	}, nil
@@ -169,11 +181,11 @@ func (c *Client) IsOnline() bool {
 }
 
 // MarkOffline - will mark a client as being offline and spawns
-// a goroutine that will attempt to reconnect if a HealthCheckPath is set.
+// a goroutine that will attempt to reconnect if HealthCheckFn is set.
 func (c *Client) MarkOffline() {
 	// Start goroutine that will attempt to reconnect.
 	// If server is already trying to reconnect this will have no effect.
-	if len(c.HealthCheckPath) > 0 && atomic.CompareAndSwapInt32(&c.connected, online, offline) {
+	if c.HealthCheckFn != nil && atomic.CompareAndSwapInt32(&c.connected, online, offline) {
 		if c.httpIdleConnsCloser != nil {
 			c.httpIdleConnsCloser()
 		}
@@ -184,12 +196,7 @@ func (c *Client) MarkOffline() {
 				if status := atomic.LoadInt32(&c.connected); status == closed {
 					return
 				}
-				ctx, cancel := context.WithTimeout(context.Background(), c.HealthCheckTimeout)
-				respBody, err := c.CallWithContext(ctx, c.HealthCheckPath, nil, nil, -1)
-				xhttp.DrainBody(respBody)
-				cancel()
-				var ne *NetworkError
-				if !errors.Is(err, context.DeadlineExceeded) && !errors.As(err, &ne) {
+				if c.HealthCheckFn() {
 					atomic.CompareAndSwapInt32(&c.connected, offline, online)
 					return
 				}

--- a/cmd/storage-rest-client.go
+++ b/cmd/storage-rest-client.go
@@ -22,14 +22,17 @@ import (
 	"crypto/tls"
 	"encoding/gob"
 	"encoding/hex"
+	"errors"
 	"io"
 	"io/ioutil"
 	"net/url"
 	"path"
 	"strconv"
 	"strings"
+	"time"
 
 	"github.com/minio/minio/cmd/http"
+	xhttp "github.com/minio/minio/cmd/http"
 	"github.com/minio/minio/cmd/logger"
 	"github.com/minio/minio/cmd/rest"
 	xnet "github.com/minio/minio/pkg/net"
@@ -656,6 +659,15 @@ func newStorageRESTClient(endpoint Endpoint) *storageRESTClient {
 	if err != nil {
 		logger.Fatal(err, "Unable to initialize remote REST disks")
 	}
-	restClient.HealthCheckPath = "/"
+
+	restClient.HealthCheckInterval = 500 * time.Millisecond
+	restClient.HealthCheckFn = func() bool {
+		ctx, cancel := context.WithTimeout(GlobalContext, restClient.HealthCheckTimeout)
+		respBody, err := restClient.CallWithContext(ctx, storageRESTMethodHealth, nil, nil, -1)
+		xhttp.DrainBody(respBody)
+		cancel()
+		return !errors.Is(err, context.DeadlineExceeded) && toStorageErr(err) != errDiskNotFound
+	}
+
 	return &storageRESTClient{endpoint: endpoint, restClient: restClient}
 }

--- a/cmd/storage-rest-common.go
+++ b/cmd/storage-rest-common.go
@@ -23,6 +23,7 @@ const (
 )
 
 const (
+	storageRESTMethodHealth               = "/health"
 	storageRESTMethodDiskInfo             = "/diskinfo"
 	storageRESTMethodCrawlAndGetDataUsage = "/crawlandgetdatausage"
 	storageRESTMethodMakeVol              = "/makevol"

--- a/cmd/storage-rest-server.go
+++ b/cmd/storage-rest-server.go
@@ -47,7 +47,11 @@ type storageRESTServer struct {
 }
 
 func (s *storageRESTServer) writeErrorResponse(w http.ResponseWriter, err error) {
-	w.WriteHeader(http.StatusForbidden)
+	if errors.Is(err, errDiskStale) {
+		w.WriteHeader(http.StatusPreconditionFailed)
+	} else {
+		w.WriteHeader(http.StatusForbidden)
+	}
 	w.Write([]byte(err.Error()))
 	w.(http.Flusher).Flush()
 }
@@ -116,6 +120,11 @@ func (s *storageRESTServer) IsValid(w http.ResponseWriter, r *http.Request) bool
 	}
 	s.writeErrorResponse(w, errDiskStale)
 	return false
+}
+
+// HealthHandler handler checks if disk is stale
+func (s *storageRESTServer) HealthHandler(w http.ResponseWriter, r *http.Request) {
+	s.IsValid(w, r)
 }
 
 // DiskInfoHandler - returns disk info.
@@ -828,6 +837,7 @@ func registerStorageRESTHandlers(router *mux.Router, endpointZones EndpointZones
 
 			subrouter := router.PathPrefix(path.Join(storageRESTPrefix, endpoint.Path)).Subrouter()
 
+			subrouter.Methods(http.MethodPost).Path(storageRESTVersionPrefix + storageRESTMethodHealth).HandlerFunc(httpTraceHdrs(server.HealthHandler))
 			subrouter.Methods(http.MethodPost).Path(storageRESTVersionPrefix + storageRESTMethodDiskInfo).HandlerFunc(httpTraceHdrs(server.DiskInfoHandler))
 			subrouter.Methods(http.MethodPost).Path(storageRESTVersionPrefix + storageRESTMethodCrawlAndGetDataUsage).HandlerFunc(httpTraceHdrs(server.CrawlAndGetDataUsageHandler))
 			subrouter.Methods(http.MethodPost).Path(storageRESTVersionPrefix + storageRESTMethodMakeVol).HandlerFunc(httpTraceHdrs(server.MakeVolHandler)).Queries(restQueries(storageRESTVolume)...)

